packages/office-ui-fabric-react/src/components/Check/Check.base.tsx

```diff
@@ -14,7 +14,11 @@ export class CheckBase extends BaseComponent<ICheckProps, {}> {
   };
 
   public shouldComponentUpdate(newProps: ICheckProps): boolean {
-    return this.props.checked !== newProps.checked;
+    return (
+      this.props.className !== newProps.className ||
+      this.props.checked !== newProps.checked ||
+      this.props.theme !== newProps.theme
+    );
```

Comment:
What if styles prop changes? I feel like having shouldComponentUpdate for such a simple component is code smell and we're checking most of the props now. I'm curious why it exists at all